File: node.py
class Node:
    def __init__(self, p, n, v):
        self.__parent = p
        self.__name = n
        self.__value = v  # (total value, time visited)
        self.__childrenList = []
        self.__endPoint = False

    def getParent(self):
        return self.__parent

    def getName(self):
        return self.__name

    def getValue(self):
        return self.__value

    def getChildrenList(self):
        return self.__childrenList

    def getEndPoint(self):
        return self.__endPoint

    def printChildrenList(self):
        s = ""
        for each in self.__childrenList:
            s += str(each.getName())
            s += ", "
        return s

    def setParent(self, p):
        self.__parent = p

    def setName(self, n):
        self.__name = n

    def setValue(self, v):
        self.__value = v

    def setChild(self, c):
        self.__childrenList.append(c)

    def setEndPoint(self, tf):
        self.__endPoint = tf

File: modules/treeStructureVersion/node.js
export class Node {
    constructor(p, n, v) {
        this._parent = p;
        this._name = n;
        this._value = v;
        this._childrenList = [];
    }
    get parent() {
        return this._parent;
    }
    get name() {
        return this._name;
    }
    get value() {
        return this._value;
    }
    get childrenList() {
        return this._childrenList;
    }
    childrenListString() {
        let s = "";
        this._childrenList.forEach(each => {
            s += each.name.toString();
            s += ", ";
        });
        return s;
    }
    set parent(p) {
        this._parent = p;
    }
    set name(n) {
        this._name = n;
    }
    set value(v) {
        this._value = v;
    }
    appendChild(c) {
        this._childrenList.push(c);
    }
}
